File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/view/UseCaseImpl.kt
package dev.hebei.camerax_android.view

import dev.hebei.camerax_android.UseCaseApi

val UseCaseApi.impl
    get() = when (this) {
        UseCaseApi.IMAGE_CAPTURE -> androidx.camera.view.CameraController.IMAGE_CAPTURE
        UseCaseApi.IMAGE_ANALYSIS -> androidx.camera.view.CameraController.IMAGE_ANALYSIS
        UseCaseApi.VIDEO_CAPTURE -> androidx.camera.view.CameraController.VIDEO_CAPTURE
    }

File: camerax_android/lib/src/view/use_case_impl.dart
import 'package:camerax_android/src/camerax.g.dart';
import 'package:camerax_platform_interface/camerax_platform_interface.dart';

extension UseCaseX on UseCase {
  UseCaseApi get api {
    switch (this) {
      case UseCase.imageCapture:
        return UseCaseApi.imageCapture;
      case UseCase.imageAnalysis:
        return UseCaseApi.imageAnalysis;
      case UseCase.videoCapture:
        return UseCaseApi.videoCapture;
    }
  }
}
